lib/capybara/poltergeist/client/compiled/web_page.js

```diff
@@ -107,16 +109,27 @@ Poltergeist.WebPage = (function() {
     });
   };
 
-  WebPage.prototype.onResourceRequestedNative = function(request) {
-    this.lastRequestId = request.id;
-    if (request.url === this.redirectURL) {
-      this.redirectURL = null;
-      this.requestId = request.id;
+  WebPage.prototype.onResourceRequestedNative = function(request, net) {
+    var abort, _ref2;
+    abort = this.urlBlacklist.some(function(blacklisted_url) {
+      return request.url.indexOf(blacklisted_url) !== -1;
+    });
+    if (abort) {
+      if (_ref2 = request.url, __indexOf.call(this._blockedUrls, _ref2) < 0) {
+        this._blockedUrls.push(request.url);
+      }
+      return net.abort();
```

Comment:
Great! I have to take a look first. Thanks for your work guys and everyone who participated!